source/MaterialXGenShader/ShaderNode.h

```diff
@@ -144,9 +144,23 @@ class ShaderInput : public ShaderPort
     /// Get optional channels value
     const string& getChannels() const { return _channels; }
 
+    /// Set isBindInput value
+    void setIsBindInput(bool isBindInput) { _isBindInput = isBindInput; }
+
+    /// Get isBindInput value
+    const bool& getIsBindInput() const { return _isBindInput;  }
+    
+    /// Set isBindParam value
+    void setIsBindParam(bool isBindParam) { _isBindParam = isBindParam; }
+
+    /// Get isBindParam value
+    const bool& getIsBindParam() const { return _isBindParam; }
+
   protected:
     ShaderOutput* _connection;
     string _channels;
+    bool _isBindInput;
```

Comment:
There there is no difference in logic between these to, I think we can just keep isBindInput and remove isBindParam. Then update the comments to say that it's either a <bindinput> or <bindparam> on a <shaderref>..